torch/csrc/autograd/variable.cpp

```diff
@@ -86,8 +86,13 @@ std::shared_ptr<Function> Variable::Impl::get_grad_accumulator() {
   std::lock_guard<std::mutex> lock(mutex_);
 
   auto result = grad_accumulator_.lock();
-  if (result)
+  const auto cur_device = data_.is_cuda() ? data_.get_device() : -1;
+  if (result && result->input_metadata(0).device() == cur_device) {
     return result;
+  } else {
+    // Resets accumulator due to out of date metadata
+    grad_accumulator_.reset();
+  }
```

Comment:
When would that happen? Can you please expand the comment?

Reply:
I believe this can occur, for example, if you call setData. We can definitely expand the comment.